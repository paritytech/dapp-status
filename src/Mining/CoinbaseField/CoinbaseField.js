// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { inject, observer } from 'mobx-react';
import Form from 'semantic-ui-react/dist/commonjs/collections/Form';
import IdentityIcon from '@parity/ui/lib/IdentityIcon';
import { injectIntl, intlShape } from 'react-intl';

import styles from './CoinbaseField.css';

class CoinbaseField extends Component {
  static propTypes = {
    accountsStore: PropTypes.object.isRequired,
    coinbaseStore: PropTypes.object.isRequired,
    intl: intlShape
  };

  handleCoinbaseChange = (_, { value }) => {
    this.props.coinbaseStore.setAuthor(value);
  };

  render() {
    const {
      accountsStore: { accounts },
      coinbaseStore: { coinbase },
      intl: { formatMessage }
    } = this.props;

    return (
      <Form.Select
        icon={
          coinbase && (
            <IdentityIcon
              address={coinbase}
              alt={coinbase}
              className={styles.coinbaseAvatar}
            />
          )
        }
        label={formatMessage(messages.coinbaseLabel)}
        onChange={this.handleCoinbaseChange}
        options={accounts.map(account => ({
          image: <IdentityIcon address={account} alt={account} />,
          key: account,
          value: account,
          text: account
        }))}
        value={coinbase}
      />
    );
  }
}

export default injectIntl(
  inject('accountsStore', 'coinbaseStore')(observer(CoinbaseField))
);

const messages = {
  coinbaseLabel: {
    id: 'dapp.status.mining.coinbaseLabel',
    defaultMessage: 'Mining Author'
  }
};
