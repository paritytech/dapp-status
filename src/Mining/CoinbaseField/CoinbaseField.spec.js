// Copyright 2015-2017 Parity Technologies (UK) Ltd.
// This file is part of Parity.

// Parity is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// Parity is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with Parity.  If not, see <http://www.gnu.org/licenses/>.

import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Form from 'semantic-ui-react/dist/commonjs/collections/Form';

import { shallowWithIntl } from '../../setupTests';
import { CoinbaseField } from './CoinbaseField';

const props = {
  accountsStore: {
    accounts: [
      '0xC257274276a4E539741Ca11b590B9447B26A8051',
      '0x1f73b8e1942165739516a93b25eac7ff1920dc1b'
    ]
  },
  coinbaseStore: {
    coinbase: '0x1f73b8e1942165739516a93b25eac7ff1920dc1b',
    setAuthor: jest.fn()
  }
};

test('should render correctly', () => {
  const component = shallowWithIntl(<CoinbaseField {...props} />);

  expect(shallowToJson(component)).toMatchSnapshot();
});

test('should handle setAuthor', () => {
  const component = shallowWithIntl(<CoinbaseField {...props} />);
  const input = component.find(Form.Select);
  input.props().onChange(null, { value: 'Foo' });

  expect(props.coinbaseStore.setAuthor).toHaveBeenCalledWith('Foo');
});
