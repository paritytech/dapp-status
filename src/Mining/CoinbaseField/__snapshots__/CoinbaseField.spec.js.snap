// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<FormSelect
  as={[Function]}
  control={[Function]}
  icon={
    <IdentityIcon
      address="0x1f73b8e1942165739516a93b25eac7ff1920dc1b"
      alt="0x1f73b8e1942165739516a93b25eac7ff1920dc1b"
      className={undefined}
    />
  }
  label="Mining Author"
  onChange={[Function]}
  options={
    Array [
      Object {
        "image": <IdentityIcon
          address="0xC257274276a4E539741Ca11b590B9447B26A8051"
          alt="0xC257274276a4E539741Ca11b590B9447B26A8051"
    />,
        "key": "0xC257274276a4E539741Ca11b590B9447B26A8051",
        "text": "0xC257274276a4E539741Ca11b590B9447B26A8051",
        "value": "0xC257274276a4E539741Ca11b590B9447B26A8051",
      },
      Object {
        "image": <IdentityIcon
          address="0x1f73b8e1942165739516a93b25eac7ff1920dc1b"
          alt="0x1f73b8e1942165739516a93b25eac7ff1920dc1b"
    />,
        "key": "0x1f73b8e1942165739516a93b25eac7ff1920dc1b",
        "text": "0x1f73b8e1942165739516a93b25eac7ff1920dc1b",
        "value": "0x1f73b8e1942165739516a93b25eac7ff1920dc1b",
      },
    ]
  }
  value="0x1f73b8e1942165739516a93b25eac7ff1920dc1b"
/>
`;
